lib/Runtime/PlatformAgnostic/Platform/Common/Intl.cpp

```diff
@@ -654,10 +684,222 @@ namespace Intl
+    // LDML here means the Unicode Locale Data Markup Language: http://www.unicode.org/reports/tr35/tr35-dates.html#Date_Field_Symbol_Table
+    // Returns the number of characters written into pattern (including a null terminator) [should always be positive]
+    int GetPatternForSkeleton(_In_z_ const char *langtag, _In_z_ const char16 *skeleton, _Out_writes_opt_(patternLen) char16 *pattern, _In_ int patternLen)
+    {
+        UErrorCode status = U_ZERO_ERROR;
+        char localeID[ULOC_FULLNAME_CAPACITY] = { 0 };
+        int length = 0;
+        uloc_forLanguageTag(langtag, localeID, ULOC_FULLNAME_CAPACITY, &length, &status);
+        ICU_ASSERT(status, length > 0);
+
+        UDateTimePatternGenerator *dtpg = udatpg_open(localeID, &status);
+        ICU_ASSERT(status, true);
+
+        int bestPatternLen = udatpg_getBestPatternWithOptions(
+            dtpg,
+            reinterpret_cast<const UChar *>(skeleton),
+            -1,
+            UDATPG_MATCH_ALL_FIELDS_LENGTH,
+            reinterpret_cast<UChar *>(pattern),
+            patternLen,
+            &status
+        );
+
+        if (pattern == nullptr && patternLen == 0 && status == U_BUFFER_OVERFLOW_ERROR)
+        {
+            // when we are just counting bytes, we can ignore errors
+            AssertOrFailFast(bestPatternLen > 0);
+        }
+        else
+        {
+            ICU_ASSERT(status, bestPatternLen > 0 && bestPatternLen < patternLen);
+        }
+
+        udatpg_close(dtpg);
+        return bestPatternLen + 1;
+    }
+
+    // creates a UDateFormat and wraps it in an IPlatformAgnosticResource
+    void CreateDateTimeFormat(_In_z_ const char *langtag, _In_z_ const char16 *timeZone, _In_z_ const char16 *pattern, _Out_ IPlatformAgnosticResource **resource)
+    {
+        UErrorCode status = U_ZERO_ERROR;
+        char localeID[ULOC_FULLNAME_CAPACITY] = { 0 };
+        int length = 0;
+        uloc_forLanguageTag(langtag, localeID, ULOC_FULLNAME_CAPACITY, &length, &status);
+        ICU_ASSERT(status, length > 0);
+
+        UDateFormat *dtf = udat_open(
+            UDAT_PATTERN,
+            UDAT_PATTERN,
+            localeID,
+            reinterpret_cast<const UChar *>(timeZone),
+            -1,
+            reinterpret_cast<const UChar *>(pattern),
+            -1,
+            &status
+        );
+        IPlatformAgnosticResource *formatterResource = new IcuCObject<UDateFormat>(dtf, &udat_close);
+        AssertOrFailFast(formatterResource);
+        *resource = formatterResource;
+    }
+
+    // Formats `date` using the UDateFormat wrapped by `resource`. If `formatted` is non-null, the result is written into `formatted`
+    // Returns the number of characters written into `formatted` (including a null terminator) [should always be positive]
+    int FormatDateTime(_In_ IPlatformAgnosticResource *resource, _In_ double date, _Out_writes_opt_(formattedLen) char16 *formatted, _In_ int formattedLen)
```

Comment:
>_In_  [](start = 23, length = 5)

maybe add const

Reply:
Remembered why I didn't do this -- it would require GetInstance to be `const TResource *GetInstance() const`, which I think can coexist with the existing GetInstance definition, but... doing more work on these classes violates my crusade against them :)